discovery_engine_core/core/src/document.rs

```diff
@@ -222,6 +223,23 @@ pub(crate) fn document_from_article(
     })
 }
 
+/// Discards documents where the title is duplicated, keeping only the top ranked.
+pub(crate) fn dedup_documents(documents: &mut Vec<Document>) {
+    documents.sort_unstable_by(|doc1, doc2| {
+        match doc1
+            .resource
+            .title
+            .to_lowercase()
+            .cmp(&doc2.resource.title.to_lowercase())
```

Comment:
We should be able to use `then_with` or `then` instead of the explicit match

Reply:
thanks for pointing that out